clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -4290,6 +4290,37 @@ class CIRIsFPClassOpLowering
   }
 };
 
+class CIRPtrMaskOpLowering : public mlir::OpConversionPattern<cir::PtrMaskOp> {
+public:
+  using OpConversionPattern<cir::PtrMaskOp>::OpConversionPattern;
+
+  mlir::LogicalResult
+  matchAndRewrite(cir::PtrMaskOp op, OpAdaptor adaptor,
+                  mlir::ConversionPatternRewriter &rewriter) const override {
+    // FIXME: We'd better to lower to mlir::LLVM::PtrMaskOp if it exists.
+    // So we have to make it manually here by following:
+    // https://llvm.org/docs/LangRef.html#llvm-ptrmask-intrinsic
```

Comment:
After this land, can you crate an issue in MLIR and another one in CIR so we can track that? Recent example: https://github.com/llvm/llvm-project/issues/115390

Reply:
Sure

Reply:
Thanks!